src/plugins/validation/2and3/semantic-validators/responses.js

```diff
@@ -10,18 +10,93 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
+              for (let i = 0; i < mediaType.schema.oneOf.length; i++) {
+                const hasInlineSchema =
+                  mediaType.schema &&
+                  mediaType.schema.oneOf &&
+                  !Object.keys(mediaType.schema.oneOf[i]).includes('$ref');
+                if (hasInlineSchema) {
+                  const checkStatus = config.inline_response_schema;
+                  if (checkStatus !== 'off') {
+                    result[checkStatus].push({
+                      path: [
+                        ...path,
+                        responseKey,
+                        'content',
+                        mediaTypeKey,
+                        'schema',
+                        'oneOf'
+                      ],
+                      message: INLINE_SCHEMA_MESSAGE
+                    });
+                  }
+                }
+              }
+            } else if (
+              mediaType &&
+              mediaType.schema &&
+              mediaType.schema.allOf
+            ) {
+              for (let i = 0; i < mediaType.schema.allOf.length; i++) {
+                const hasInlineSchema =
+                  mediaType.schema &&
+                  mediaType.schema.allOf &&
+                  !Object.keys(mediaType.schema.allOf[i]).includes('$ref');
+                if (hasInlineSchema) {
+                  const checkStatus = config.inline_response_schema;
+                  if (checkStatus !== 'off') {
+                    result[checkStatus].push({
+                      path: [
+                        ...path,
+                        responseKey,
+                        'content',
+                        mediaTypeKey,
+                        'schema',
+                        'allOf'
+                      ],
+                      message: INLINE_SCHEMA_MESSAGE
+                    });
+                  }
+                }
+              }
+            } else if (
+              mediaType &&
+              mediaType.schema &&
+              mediaType.schema.anyOf
+            ) {
+              for (let i = 0; i < mediaType.schema.anyOf.length; i++) {
+                const hasInlineSchema =
+                  mediaType.schema &&
+                  mediaType.schema.anyOf &&
+                  !Object.keys(mediaType.schema.anyOf[i]).includes('$ref');
+                if (hasInlineSchema) {
+                  const checkStatus = config.inline_response_schema;
+                  if (checkStatus !== 'off') {
+                    result[checkStatus].push({
+                      path: [
+                        ...path,
+                        responseKey,
+                        'content',
+                        mediaTypeKey,
+                        'schema',
+                        'anyOf'
+                      ],
+                      message: INLINE_SCHEMA_MESSAGE
+                    });
+                  }
+                }
+              }
```

Comment:
Will you refactor these checks so that you only need the code once? Something like
```js
const combinedSchemaTypes = ['allOf', 'oneOf', 'anyOf'];
combinedSchemaTypes.forEach(schemaType => {
  // ...
});
```